auth/oidc.go

```diff
@@ -252,7 +266,12 @@ func (op *OIDCProvider) InitOIDCClient() error {
 		return err
 	}
 
-	op.client = &OIDCClient{Config: config, Verifier: verifier}
+	op.client = &OIDCClient{Config: config}
+	op.client.SetVerifier(verifier)
+
+	sync := newProviderConfigSync()
+	sync.config = op
+	sync.Run()
```

Comment:
sync.Run() returns a channel that should be closed when the Sync Gateway database is stopped, in order to close the sync goroutines.  

There are two approaches we could use here:
- have the databaseContext pass in it's terminator, which gets closed on dbContext.Close()
- add an OIDCProvider.Stop() method that closes the channel returned by sync.Run(), and call OIDCProvider.Stop() inside dbContext.Close()